processor/spanmetricsprocessor/processor.go

```diff
@@ -322,54 +354,82 @@ func (p *processorImp) aggregateMetricsForSpan(serviceName string, span pdata.Sp
 	// Binary search to find the latencyInMilliseconds bucket index.
 	index := sort.SearchFloat64s(p.latencyBounds, latencyInMilliseconds)
 
-	key := buildKey(serviceName, span, p.dimensions, resourceAttr)
+	mKey := p.buildMetricKey(span, resourceAttr)
+	resourceAttrKey := p.buildResourceAttrKey(serviceName, resourceAttr)
+	p.resourceAttrList[resourceAttrKey] = true
 
 	p.lock.Lock()
-	p.cache(serviceName, span, key, resourceAttr)
-	p.updateCallMetrics(key)
-	p.updateLatencyMetrics(key, latencyInMilliseconds, index)
-	p.updateLatencyExemplars(key, latencyInMilliseconds, span.TraceID())
+	p.cacheMetricKey(span, mKey, resourceAttr)
+	p.cacheResourceAttrKey(serviceName, resourceAttr, resourceAttrKey)
+	p.updateCallMetrics(resourceAttrKey, mKey)
+	p.updateLatencyMetrics(resourceAttrKey, mKey, latencyInMilliseconds, index)
+	p.updateLatencyExemplars(resourceAttrKey, mKey, latencyInMilliseconds, span.TraceID())
 	p.lock.Unlock()
 }
 
 // updateCallMetrics increments the call count for the given metric key.
-func (p *processorImp) updateCallMetrics(key metricKey) {
-	p.callSum[key]++
+func (p *processorImp) updateCallMetrics(resourceAttrKey resourceKey, mKey metricKey) {
+	if _, ok := p.callSum[resourceAttrKey]; ok {
+		p.callSum[resourceAttrKey][mKey]++
+	} else {
+		p.callSum[resourceAttrKey] = map[metricKey]int64{mKey: 1}
+	}
 }
 
 // updateLatencyExemplars sets the histogram exemplars for the given metric key and append the exemplar data.
-func (p *processorImp) updateLatencyExemplars(key metricKey, value float64, traceID pdata.TraceID) {
-	if _, ok := p.latencyExemplarsData[key]; !ok {
-		p.latencyExemplarsData[key] = []exemplarData{}
+func (p *processorImp) updateLatencyExemplars(resourceAttrKey resourceKey, mKey metricKey, value float64, traceID pdata.TraceID) {
+	if _, ok := p.latencyExemplarsData[resourceAttrKey]; ok {
+		if _, ok := p.latencyExemplarsData[resourceAttrKey][mKey]; !ok {
+			p.latencyExemplarsData[resourceAttrKey][mKey] = []exemplarData{}
+		}
+	} else {
+		p.latencyExemplarsData[resourceAttrKey] = make(map[metricKey][]exemplarData)
+		p.latencyExemplarsData[resourceAttrKey][mKey] = []exemplarData{}
```

Comment:
If you flip the condition here, you can save your if else statement.

```suggestion
_, ok := p.latencyExamplarsData[p.resourceAttrKey]
if !ok {
    p.latencyExamplarsData[p.resourceAttrKey] = make(map[metricKey][]examplarData)
    p.latencyExamplarsData[p.resourceAttrKey][mkey] = make([]examplarData)
}

if _, ok = p.latencyExamplarsData[p.resourceAttrKey][mkey]; !ok {
   p.latencyExamplarsData[p.resourceAttrKey][mkey] = make([]examplarData)
}
```

Just avoids nested if statements and keeps performance consistent across conditions.